src/enzyme_ad/jax/Dialect/EnzymeXLAOps.td

```diff
@@ -61,12 +61,13 @@ def KernelCallOp: EnzymeXLA_Op<"kernel_call", [DeclareOpInterfaceMethods<SymbolU
   let hasCanonicalizer = 1;
 }
 
-def JITCallOp: EnzymeXLA_Op<"jit_call", [DeclareOpInterfaceMethods<SymbolUserOpInterface>, DeclareOpInterfaceMethods<CallOpInterface>, Pure]> {
+def JITCallOp: EnzymeXLA_Op<"jit_call", [DeclareOpInterfaceMethods<SymbolUserOpInterface>, DeclareOpInterfaceMethods<CallOpInterface>, DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
   let summary = "JIT Call operation";
 
   let arguments = (ins
-    FlatSymbolRefAttr:$fn, 
+    FlatSymbolRefAttr:$fn,
     Variadic<AnyType>:$inputs,
+    DefaultValuedOptionalAttr<BoolAttr, "false">:$has_side_effect,
```

Comment:
Nit: you can use `UnitAttr` the presence of which indicates "true" and the absence "false" instead to avoid the annoying ternary state.